packages/office-ui-fabric-react/src/components/Button/_base/Button.base.types.ts

```diff
@@ -0,0 +1,183 @@
+import * as React from 'react';
+import { ButtonBase } from './Button.base';
+import { IRenderFunction } from '../../../Utilities';
+import { IIconProps } from '../../../Icon';
+import { IStyleFunction } from '../../../Utilities';
+import {
+  IStyle,
+  ITheme
+} from '../../../Styling';
+
+export interface IButtonBase {
+  /**
+   * Sets focus to the button.
+   */
+  focus: () => void;
+}
+
+export interface IButtonBaseProps extends React.AllHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | ButtonBase> {
+  /**
+   * Optional callback to access the IButton interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: (component: IButtonBase) => void;
+
+  /**
+   * If provided, this component will be rendered as an anchor.
+   * @default ElementType.anchor
+   */
+  href?: string;
+
+  /**
```

Comment:
Isn't primary more of a styling concern? Would it make more sense to go on the styling props?

Reply:
The idea is to have a single component with 2 style variants. Otherwise we'd have defaultButton, primaryButton, defaultCompound button, primaryCompound button and on and on. This also solves the issue some of had switching a button from primary to default without losing focus, as its the same component with different props.

Reply:
I think the solution to the "primary" button scenario is similar to a developer that wants to use a DefaultButton, but just change one or two CSS values (maybe the background color). How does that story look with this approach?

Reply:
That story works as expected. 
```
<DefaultButton getStyles={(props) => ({button: {background: 'blue'}) />
```
The reason I keep coming back to 'primary' or something similar to it, is that almost ALL of our buttons have 2 different themes, standard and primary, and there is a desire to be able to switch between them without losing focus.

Removing primary would mean going back to components for every style variation, which feels like...well less overhead now, but still feels like it'd clutter things up.